src/mono/mono/mini/interp/interp.c

```diff
@@ -8592,6 +8593,17 @@ metadata_update_prepare_to_invalidate (void)
 }
 
 
+static void 
+interp_invalidate_transformed_func (void* data, gpointer user_data)
+{
+	MonoAssemblyLoadContext* alc = (MonoAssemblyLoadContext*)data;
+	MonoJitMemoryManager *jit_mm = (MonoJitMemoryManager*)(alc->memory_manager->runtime_info);
+
+	jit_mm_lock (jit_mm);
```

Comment:
When `interp_invalidate_transformed_func` is called, it will take the jit memory manager lock.

So we're implicitly introducing a lock order: `alcs_list_lock > MonoMemoryManager:lock`